float64 angle2nd
